drivers/ec.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Lightweight ACPI Implementation
+ * Copyright (C) 2018-2019 the lai authors
+ */
+
+/* LAI Embedded Controller implementation
+ * ACPI 6.3 Specification chapter 12
+ * ACPI defines an Embedded Controller for interfacing directly with firmware
+ */
+
+#include <lai/drivers/ec.h>
+
+#define ACPI_EC_PNP_ID "PNP0C09"
+
+static uint16_t ec_cmd_port = 0;
+static uint16_t ec_data_port = 0;
```

Comment:
Those two variables should not be global. Make them part of a `struct lai_ec_driver` and let the functions take a `struct lai_ec_driver *`.